schema_based_on_diagram.sql

```diff
@@ -0,0 +1,46 @@
+CREATE TABLE medical_histories (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id INT,
+  status VARCHAR,
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE patients (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  name VARCHAR,
+  date_of_birth DATE,
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE treatments (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  type VARCHAR,
+  name VARCHAR,
+  PRIMARY KEY(id),
+  FOREIGN KEY (id) REFERENCES medical_histories (id)
+);
+
+CREATE TABLE invoices (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INT,
+  PRIMARY KEY(id),
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories (id)
+);
+
+CREATE TABLE invoice_items (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  FOREIGN KEY (invoice_id) REFERENCES invoices (id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments (id)
+);
+
```

Comment:
Please be sure to create a [joining table](https://dzone.com/articles/how-to-handle-a-many-to-many-relationship-in-datab#:~:text=A%20joining%20table%20is%20a,a%20much%20better%20data%20structure.) between `medical_histories` and `treatments` describing the many-to-many relationship 🤓